engine/src/main/java/org/hibernate/search/backend/impl/lucene/AbstractWorkspaceImpl.java

```diff
@@ -107,12 +115,41 @@ public IndexWriter getIndexWriter(ErrorContextBuilder errorContextBuilder) {
 
 	@Override
 	public boolean areSingleTermDeletesSafe() {
-		return indexMetadataIsComplete && entitiesInIndexManager.size() == 1;
+		return indexMetadataIsComplete && getEntitiesInIndexManager().size() == 1;
 	}
 
 	@Override
 	public boolean isDeleteByTermEnforced() {
```

Comment:
`areSingleTermDeletesSafe` should be named `isSingleClassInIndex`
`isDeleteByTermEnforced` should be named `isDeleteByTermEnforcedOrSafe`